### PR TITLE
Machine i2c deinit

### PR DESCRIPTION
**Pull request by jaenrig-ifx**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

To be reviewed after #95.

Added deinitialization function to i2c module. This is directly added to the extmod.

Docs amendments will be done after the complete machine module is refactored for the pin_phy.

### CODE REVIEW

File: ports/psoc6/modules/machine/machine_i2c.c [hunk 1]
```diff
@@ -173,13 +203,12 @@ STATIC int machine_i2c_transfer(mp_obj_base_t *self_in, uint16_t addr, size_t le
     }
 }
 
-
 STATIC const mp_machine_i2c_p_t machine_i2c_p = {
+    .deinit = machine_i2c_deinit,
```
Comment (NikhitaR-IFX) [inline ports/psoc6/modules/machine/machine_i2c.c:207]: Not completely sure if this is needed. So far I understand, this is to point to our version of implementation of function, if there is already one implemented in other module (for ex. in extmod/i2c). And I think there is not deinit in extmod. But I might be wrong and feel free to correct.

File: ports/psoc6/modules/machine/machine_i2c.c [hunk 2]
```diff
@@ -189,3 +218,11 @@ MP_DEFINE_CONST_OBJ_TYPE(
     protocol, &machine_i2c_p,
     locals_dict, &mp_machine_i2c_locals_dict
     );
+
+void mod_i2c_deinit() {
```
Comment (NikhitaR-IFX) [inline ports/psoc6/modules/machine/machine_i2c.c:222]: Also handles for softI2C?

**Review by NikhitaR-IFX — APPROVED** [review]

File: ports/psoc6/modules/machine/machine_i2c.c (hunk 1 above)
Comment (jaenrig-ifx) [reply ports/psoc6/modules/machine/machine_i2c.c:207]: Correct. That was not declared in the extmod. And I added it. 
How can the i2c then be deinited? I wonder how other ports handle softreset: if there is no deinit(), after a softreset the i2c is still configured in whatever mode? If we want to reuse? we reinitiaze? with no impact? 

As discuss, I don´t yet understand how the garbage collector handles destructing all objects... but I don´t see (yet) the connection to hardware peripheral deinitialization. That has to be handled somewhere (freeing before initing() all the time?)...

**Review by IFX-Anusha — APPROVED** [review]

File: ports/psoc6/modules/machine/machine_i2c.c (hunk 2 above)
Comment (IFX-Anusha) [reply ports/psoc6/modules/machine/machine_i2c.c:222]: GPIO initialization for the softI2C is handled inside mphalport.c using cyhal_gpio_configure( ) function

File: ports/psoc6/modules/machine/machine_i2c.c [hunk 3]
```diff
@@ -15,26 +15,46 @@
 
 // port-specific includes
 #include "modmachine.h"
+#include "machine_pin_phy.h"
 #include "mplogger.h"
 
+#define MAX_I2C              10 // TODO: Change as per bsp?
 #define DEFAULT_I2C_FREQ     (400000)
-#define MICROPY_HW_I2C_SCL   (CYBSP_I2C_SCL)
```
[the remark this answers is not in the document]
Comment (jaenrig-ifx) [reply ports/psoc6/modules/machine/machine_i2c.c:21]: Yes, I removed that. There is no such thing as "default" pins unless we provide pre-instantiated objects to those default pins allocation... and that has to be documented in a pinout diagram or table.
For multiple board, this is harder to maintain. But we can have such default allocation, then that comes with a nice pin out diagram and we have to add those to the code. But then the user can happily call i2c =  I2C(0) for any board.
In the end, the user needs to know the pins, so we can skip the magic of not providing any (at least for now).

File: ports/psoc6/modules/machine/machine_i2c.c (hunk 2 above)
Comment (jaenrig-ifx) [reply ports/psoc6/modules/machine/machine_i2c.c:222]: Mmm okay, I think then we have to extend that to initialize that pin if not inited first.

File: ports/psoc6/modules/machine/machine_i2c.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (IFX-Anusha) [reply ports/psoc6/modules/machine/machine_i2c.c:21]: okay. sounds good

**Pull request merged**